### ISSUE TITLE
Possibility to have saved models including their optimizer and scheduler state_dicts

### ISSUE THREAD

**Issue by g12bftd** [pytorch/vision#7740]
### 🚀 The feature

In PyTorch, it is possible to save a model along with all the entities involved in training (optimiser, scheduler) etc. Would it be possible to have these for ResNet-18,34 and 50?

### Motivation, pitch

I am working with the Brevitas library to quantise models and re-train them (NOT from scratch but using the pre-trained models). However, when re-training from a floating point model, I noticed it's best to have access to optimiser and scheduler state dictionaries. Otherwise the training is sub-optimal.
